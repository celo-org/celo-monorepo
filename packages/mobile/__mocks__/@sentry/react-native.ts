const Sentry = {
  config: () => ({ install: jest.fn() }),
  setTagsContext: jest.fn(),
  captureException: jest.fn(),
}

module.exports = Sentry
