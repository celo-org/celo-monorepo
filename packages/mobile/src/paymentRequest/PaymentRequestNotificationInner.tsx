import fontStyles from '@celo/react-components/styles/fonts'
import * as React from 'react'
import { StyleSheet, Text } from 'react-native'
import { CURRENCIES, CURRENCY_ENUM } from 'src/geth/consts'
import { Recipient } from 'src/recipients/recipient'
import { getCentAwareMoneyDisplay } from 'src/utils/formatting'

interface Props {
  requesterE164Number: string
  comment: string
  amount: string
  requesterRecipient: Recipient | null
}

export default function PaymentRequestNotificationInner(props: Props) {
  const { requesterE164Number, comment: message, amount, requesterRecipient } = props
  return (
    <Text numberOfLines={1} ellipsizeMode="middle" style={styles.oneLine}>
      <Text style={[fontStyles.subSmall]}>
        {(requesterRecipient && requesterRecipient.displayName) || requesterE164Number} - {message}
      </Text>
      <Text style={[fontStyles.subSmall, fontStyles.semiBold]}>
        {' ' + CURRENCIES[CURRENCY_ENUM.DOLLAR].symbol + getCentAwareMoneyDisplay(amount)}
      </Text>
    </Text>
  )
}

const styles = StyleSheet.create({
  oneLine: {
    flexDirection: 'row',
  },
})
