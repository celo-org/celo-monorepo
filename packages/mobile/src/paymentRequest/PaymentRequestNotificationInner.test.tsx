import * as React from 'react'
import 'react-native'
import * as renderer from 'react-test-renderer'
import PaymentRequestNotificationInner from 'src/paymentRequest/PaymentRequestNotificationInner'

it('renders correctly', () => {
  const tree = renderer.create(
    <PaymentRequestNotificationInner
      requesterE164Number="+14155552671"
      amount="24"
      requesterRecipient={null}
    />
  )

  expect(tree).toMatchSnapshot()
})
