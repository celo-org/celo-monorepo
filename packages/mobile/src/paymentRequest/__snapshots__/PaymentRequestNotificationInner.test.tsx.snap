// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  ellipsizeMode="middle"
  numberOfLines={1}
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#81868B",
        "fontFamily": "Hind-Regular",
        "fontSize": 14,
        "lineHeight": 20,
      }
    }
  >
    +14155552671
     for 
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#81868B",
          "fontFamily": "Hind-Regular",
          "fontSize": 14,
          "lineHeight": 20,
        },
        Object {
          "fontFamily": "Hind-SemiBold",
        },
      ]
    }
  >
    24
  </Text>
</Text>
`;
